enclave_build/src/docker.rs

```diff
@@ -205,6 +205,15 @@ impl DockerUtil {
                             }
                         }
                         Err(e) => {
+                            // Check if the Docker image is locally available and not remotely.
+                            // If not locally available return error.
+                            if let Ok(_image) =
```

Comment:
I think the behavior here might be surprising to users. Please correct me, but I believe the logic is now:

1. Attempt to pull the image.
2. If the pull fails, check if the image exists locally.

This now requires that if a user build a docker image locally it must not have a name that can be pulled, or else the EIF build _will_ pull it, overwriting their local image.

I believe the unsurprising behavior is to check if the image is locally available _first_, and if it's not _then_ attempt to pull it.

Reply:
Correct, that's how logic currently works. I was thinking of this case, just that the previous logic, before the crate update, was pulling the image anyway, but not exposing the possible error when trying to fetch from the remote repo.

Anyhow, that's a fair point, I can update the logic to check for local images first. Thanks for bringing it up.